chain/db/validator.go

```diff
@@ -88,3 +109,58 @@ func (b *BandDB) ClearOldVotes(currentHeight int64) error {
 	}
 	return nil
 }
+
+func (b *BandDB) GetValidator(validator sdk.ValAddress) (Validator, bool) {
+	validatorStruct := Validator{OperatorAddress: validator.String()}
+	err := b.tx.First(&validatorStruct).Error
+	return validatorStruct, err == nil
+}
+
+func (b *BandDB) handleMsgEditValidator(msg staking.MsgEditValidator) error {
+	validator, isFound := b.GetValidator(msg.ValidatorAddress)
+	if !isFound {
+		return fmt.Errorf(fmt.Sprintf("validator %s has not exist.", msg.ValidatorAddress.String()))
+	}
+
+	if msg.Description.Moniker != staking.DoNotModifyDesc {
+		validator.Moniker = msg.Description.Moniker
+	}
+	if msg.Description.Identity != staking.DoNotModifyDesc {
+		validator.Identity = msg.Description.Identity
+	}
+	if msg.Description.Website != staking.DoNotModifyDesc {
+		validator.Website = msg.Description.Website
+	}
+	if msg.Description.Details != staking.DoNotModifyDesc {
+		validator.Details = msg.Description.Details
+	}
+	if msg.CommissionRate != nil {
+		validator.CommissionRate = msg.CommissionRate.String()
+	}
+	if msg.MinSelfDelegation != nil {
+		validator.MinSelfDelegation = msg.MinSelfDelegation.ToDec().String()
+	}
+
+	return b.tx.Save(&validator).Error
+}
+
+func (b *BandDB) handleMsgCreateValidator(msg staking.MsgCreateValidator) error {
+	_, isFound := b.GetValidator(msg.ValidatorAddress)
+	if isFound {
+		return fmt.Errorf(fmt.Sprintf("validator %s has already exist.", msg.ValidatorAddress.String()))
```

Comment:
Why do we need to do this? I don't think it's even possible to not found.

Reply:
to found ?